### DISCUSSION TITLE
Hot load at runtime

### DISCUSSION THREAD

**Discussion by wotori** [bevyengine/bevy#4184] [Q&A] [answered]
Hello! I want to creat a gallery of 3D assets based on Bevy which hot-load it from remote server, IPFS or folder on computer and add to the scene. Could anybody point me, how can I do this and is it possible at all?

**Comment by marko-lazic** [comment, marked as answer]
Create a backend piece of software that scans and generates a scene.scn file based on an asset in your specified directory.
It should pack .scn into a new directory together with copied assets into your 3D gallery assets/scenes folder.
Create a bevy app that would scan assets/scenes for new scenes and spawn them to the gallery.

Remarks

- There is asset_server.watch_for_changes() that may be of interest to you.
- Spawn scene with scene_spawner.spawn_dynamic()
- bevy_blender_toolkit is one place to look at how scenes can be created and assets loaded.
- Make a simple configuration in your backend app. Path to folder on the computer, path to the gallery assets folder.
- Draw some diagrams before coding. Cheep whiteboard and whiteboard markers may be cheaper and faster than diagram & flowchart editors.
- After being satisfied with the local directory try implementing it from a remote server.

**Reply by wotori** [reply]
thank you for such a detailed answer!
